### ISSUE TITLE
Shuffle Key Length Greater than 32

### ISSUE THREAD

**Issue by sguerin13** [microsoft/O-CNN#128]
Hello, I am currently using the pytorch implementation and trying to visualize the output using the shuffle keys. I call ocnn.octree_property(octree_tensor,'xyz',depth=6) and then convert the output values to a binary string. Once I have the binary string, I parse it using the encoding scheme mentioned in your paper. 

However I am seeing that the output binary strings in my case are longer than 32 - ranging from a length of 36 to 38 when my octree has a depth of 6 - and the visual results indicate that something is off. I believe my shuffle key parsing correct assuming the pattern you mention in your paper, however I believe my conversion to binary may be off. 

Here is the code snippet I use to extract the binary string:

oct_tens = pickle.load(open('octree_tensor.pt','rb'))
keys = ocnn.octree_property(oct_tens,'xyz',6)
bin_keys = [bin(i) for i in keys]

One thing I saw is that the oct_tens stores the integers as 64 bit and the values all seem to be larger than standard 32-bit integers. Perhaps this is causing the issue. 

I have included the octree_tensor for reference as well. 

[octree_tensor.zip](https://github.com/microsoft/O-CNN/files/6744977/octree_tensor.zip)

**Comment by wang-ps** [comment]
Yes, the key in the pytorch implementation is 64 bit. 
The code you used actually extracts the x, y, z coordinates, instead of the shuffled key. 

```python
    xyz = ocnn.octree_property(oct_tens,'xyz',6)  # the zipped xyz coordinates
    pts = ocnn.octree_decode_key(xyz)             # pts are the decoded coordinates

    key = ocnn.octree_property(oct_tens,'key',6)  # the shuffled key
```

I checked the octree you uploaded, and visualized the pts for your reference. The points are a little messy.
![octree](https://user-images.githubusercontent.com/7525130/124344504-3231be00-dc05-11eb-8b0b-52a205f607f3.jpg)

**Comment by sguerin13** [comment]
Thank you for providing a solution for this. I am using your model for my thesis and this helps me out a lot. 

Using the solution you described above I was still running into issues getting the output shape to match the input shape, but I figured out what else I was doing wrong. In case anyone else runs into this issue, I was generating the point object using ocnn.points_new() and reading a ply file using PyntCloud. My issue was that I converting from PyntCloud dataframe to numpy array and then to torch tensor. After looking at your points_property test I saw that you converted lists to tensors instead of numpy arrays. So I changed the tensor input to a list and that solved the problem (code below). I sanity checked myself using the stanford bunny. 

points_tensor = torch.tensor(**list**(pnts.to_numpy())).float()
features_tensor = torch.tensor([]).float()
normals_tensor = torch.tensor(**list**(pnts_normals.to_numpy())).float()
labels_tensor = torch.tensor([]).float()
OCNN_point = ocnn.points_new(points_tensor,normals_tensor,features_tensor,labels_tensor)


![bunny_reconstructed](https://user-images.githubusercontent.com/15376958/124521292-fb060d00-ddbc-11eb-8b11-11c23b869d96.png)
